_posts/2020-10-16-kaitai-struct-v0.9-released.md

```diff
@@ -72,6 +74,7 @@ fixes a lot of bugs and includes quite a few infrastructure improvements.
 * Runtime API changes:
   * Add exceptions `Validation{Not{Equal,AnyOf},{Less,Greater}Than,Expr}Error` inheriting from common ancestor `ValidationFailedError` - thrown on failed validations defined with `valid` or `contents` key ([#435](https://github.com/kaitai-io/kaitai_struct/issues/435))
   * Add method `read_bits_int_le` for parsing little-endian bit-sized integers ([docs](https://doc.kaitai.io/user_guide.html#bit-ints-le))
+  * Added an [official library of compression algorithms](https://github.com/kaitai-io/kaitai_compress) to be used in `process` and changed the compiler to allow third-party `process`ors ([#457](https://github.com/kaitai-io/kaitai_struct/issues/457))
```

Comment:
Neither _"added a library of compression algorithms"_ nor _"changed the compiler ..."_ is actually a **runtime API change**. And these are actually two independent things - the fact that the latter allows the first is not important in the changelog.

A changelog should be a highly structured file, and it is usually not read whole as a book - people will just scan it quickly. For this reason, I try to strictly group the changes with the **same type** _(in the spirit of [**Keep a changelog** principles](https://keepachangelog.com/en/1.0.0/#:~:text=The%20same,be%20grouped.))_, even if that means that points relating to one feature will be scattered in multiple groups. This can be seen on the example of the validation feature - it's splitted across _New KSY language features_ and _Runtime API changes_.

Hmm, I'm not sure though where to move the point about `kaitai_compress`. That looks like it should be in a special section, I'd say _Related projects_ at the bottom of the release notes.

Reply:
Done.